tiny_dnn/parameter.h

```diff
@@ -0,0 +1,125 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include "tiny_dnn/core/framework/tensor.h"
+#include "tiny_dnn/util/util.h"
+
+namespace tiny_dnn {
+
+enum class parameter_type : int8_t { weight = 0x0001, bias = 0x0002 };
+
+class parameter {
+ public:
+  /**
+   * Initializes an empty parameter taking in the dimensions of weights and
+   * biases of a layer. Currently supported for maximum 4-dimensions, and
+   * stored as a flat ``Tensor``.
+   *
```

Comment:
Dealing with views is not as smooth as it should be, but it should be functional within some time. Besides we are accessing members of data and grad from a flat tensor so it will help me integrate working model of parameter without making radical changes. But yes, I will add a documentation of NCHW.